### PR TITLE
feat: Allow for multiple reporting formats

### LINKED ISSUE #5: Allow for multiple reporting formats
In addition to reporting to Slack channels, it'd be nice to send out summary emails. Maybe we could even report to other systems eventually. Other users of the tool might want to report to Microsoft Teams. Or maybe we could create a reporter that automatically creates JIRA tickets. The possibilities are endless!

The first step will be create a generic `Reporter` interface, then creating a `SlackReporter` as a concrete implementation of that interface. From there we can also create an `EmailReporter` implementation.

### PR DESCRIPTION
**Pull request by tarkatronic**
Fixes #5 

* All reporting related stuff is now in the `reporting` package
  * This may require some more thought on organization. I was pretty heavy-handed here.
* There's now a `Reporter` interface
* There's a `SlackReporter` implementation of that interface
* Slack functionality is all isolated inside the Slack module
* Slack messages are sent asynchronously!

Oh also there's no tests here. Which sucks. But this PR is already so massive...

Largely, this PR is shuffling a whole lot of code around to new places.

### CODE REVIEW

**Comment by tarkatronic** [comment]
I apologize for the size of this PR. There was a lot of necessary reshuffling to make this all work right. And once I got started on tests, there was no stopping me. 😅

File: reporting/slack.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
+package reporting
```
Comment (JoseAngel1196) [inline reporting/slack.go:1]: 🤩

**Review by JoseAngel1196 — APPROVED** [review]
Amazing! This is becoming more flexible to work on. Now, I want to get some work done here. 😄

**Comment by tarkatronic** [comment]
> Amazing! This is becoming more flexible to work on. Now, I want to get some work done here. 😄

Thanks! I really hope this works out as well as we think it will.

I think I'm going to leave this PR open at least for the night, to give others a chance to pick it apart. As excited as I am to get this merged 😄

**Pull request merged**